docs/root/configuration/http/http_filters/set_filter_state.rst

```diff
@@ -0,0 +1,54 @@
+.. _config_http_filters_set_filter_state:
+
+Envoy Set-Filter-State HTTP Filter
+==================================
+* This filter should be configured with the type URL ``type.googleapis.com/envoy.extensions.filters.http.set_filter_state.v3.Config``.
+* :ref:`v3 API reference <envoy_v3_api_msg_extensions.filters.http.set_filter_state.v3.Config>`
+
+This filter is configured with rules that update the request filter state using
+the request data. The filter state value can then be used for routing, load
+balancing decisions, telemetry, etc. See :ref:`the well-known filter state keys
+<well_known_filter_state>` for the controls used by the other extensions.
+
+Examples
+--------
+
+A sample filter configuration to capture the authority header from an HTTP
+CONNECT request as the original destination cluster dynamic destination
+address:
+
+.. validated-code-block:: yaml
+  :type-name: envoy.extensions.filters.http.set_filter_state.v3.Config
+
+  rules:
+  - key: envoy.network.transport_socket.original_dst_address
```

Comment:
Oh, I completely missed that this needs to interact with "well-known" filters. This makes my comment above on `key` naming moot.

IMHO this ability makes it more challenging to understand and control the different filters behaviors, as it essentially overrides previous filters "decisions".

Reply:
Yes, this filter can be very dangerous. I'll make a big note on the docs about the capability of this filter to override other filters.